tests/Integration/Behaviour/Features/Scenario/Product/Shop/update_multishop_image.feature

```diff
@@ -0,0 +1,100 @@
+    And product "product1" should have following cover "http://myshop.com/img/p/{no_picture}-cart_default.jpg" for shops "shop3"
+    When I add new image "image1" named "app_icon.png" to product "product1" for shop "shop1"
+    And I copy product product1 from shop shop1 to shop shop3
+    When I add new image "image2" named "logo.jpg" to product "product1" for shop "shop1"
+    And I copy product product1 from shop shop1 to shop shop2
+    And product "product1" should have following images for shop "shop1, shop2":
+      | image reference | is cover | legend[en-US] | position | image url                            | thumbnail url                                      |
+      | image1          | true     |               | 1        | http://myshop.com/img/p/{image1}.jpg | http://myshop.com/img/p/{image1}-small_default.jpg |
+      | image2          | false    |               | 2        | http://myshop.com/img/p/{image2}.jpg | http://myshop.com/img/p/{image2}-small_default.jpg |
+    And product "product1" should have following images for shop "shop3":
+      | image reference | is cover | legend[en-US] | position | image url                            | thumbnail url                                      |
+      | image1          | true     |               | 1        | http://myshop.com/img/p/{image1}.jpg | http://myshop.com/img/p/{image1}-small_default.jpg |
+    And product "product1" should have following cover "http://myshop.com/img/p/{image1}-cart_default.jpg" for shops "shop1, shop2, shop3"
+    And images "[image1, image2]" should have following types generated:
+      | name           | width | height |
+      | cart_default   | 125   | 125    |
+      | home_default   | 250   | 250    |
+      | large_default  | 800   | 800    |
+      | medium_default | 452   | 452    |
+      | small_default  | 98    | 98     |
+
+  Scenario: I update image file
+    When I update image "image1" with following information for shop "shop1":
+      | file | logo.jpg |
+    Then image "image1" should have same file as "logo.jpg"
+    When I update image "image1" with following information for shop "shop2":
+      | file | app_icon.png |
+    Then image "image1" should have same file as "app_icon.png"
+
+  Scenario: I update image legend
+    When I update image "image1" with following information for shop "shop1":
+      | legend[en-US] | preston is alive |
+    Then product "product1" should have following images for shop "shop1, shop2":
+      | image reference | is cover | legend[en-US]    | position | image url                            | thumbnail url                                      |
+      | image1          | true     | preston is alive | 1        | http://myshop.com/img/p/{image1}.jpg | http://myshop.com/img/p/{image1}-small_default.jpg |
+      | image2          | false    |                  | 2        | http://myshop.com/img/p/{image2}.jpg | http://myshop.com/img/p/{image2}-small_default.jpg |
```

Comment:
You should also check the content for shop3, so we can see that legend is also updated for the other shop (since it's a shared value for all shops)